Lib/zipapp.py

```diff
@@ -73,8 +74,23 @@ def _copy_archive(archive, new_archive, interpreter=None):
         os.chmod(new_archive, os.stat(new_archive).st_mode | stat.S_IEXEC)
 
 
+def _build_filter(*patterns):
+    """
+    Given a list of patterns, return a callable that will be true only if
+    the input matches at least one of the patterns.
+    """
+    return lambda name: any(fnmatchcase(name, pat=pat) for pat in patterns)
+
+
+def default_filter(package, exclude=(), include=('*',)):
+    package = str(package)
+    exclude_filter = _build_filter(*exclude)
+    include_filter = _build_filter(*include)
+    return include_filter(package) and not exclude_filter(package)
+
+
 def create_archive(source, target=None, interpreter=None, main=None,
-                   filter=None, compressed=False):
+                   filter=None, compressed=False, **filter_kwargs):
```

Comment:
I'm concerned about file arguments for the filter being "anything that doesn't already have a meaning". What if we want to add another argument to the `create_archive` function? This functionality seems a bit too niche to "steal" all of the remaining potential argument names.

I'd prefer the `filter_kwargs` argument to be a normal argument that takes a dictionary. Or maybe even just pass a single argument to the filter - let the filter decide how to interpret it.